controllers/imagejob/imagejob_controller.go

```diff
@@ -330,6 +334,35 @@ func (r *Reconciler) handleNewJob(ctx context.Context, imageJob *eraserv1.ImageJ
 		return err
 	}
 
+	// keep track of PodRunning threads
+	var waitGroup sync.WaitGroup
+
+	// sized channel to limit number of threads at once
+	podQueue := make(chan *corev1.Pod, *podRunningThreadLimit)
+
+	// range over podQueue in background to prevent blocking
+	go func() {
+		for pod := range podQueue {
+			// start thread to wait for PodRunning
+			go func(pod *corev1.Pod) {
+				if err := wait.PollImmediate(time.Nanosecond, time.Minute*5, r.isPodRunning(pod.Name, pod.Namespace)); err != nil {
+					currentPod := &corev1.Pod{}
+					namespacedName := types.NamespacedName{
+						Namespace: pod.Namespace,
+						Name:      pod.Name,
+					}
+
+					if err := r.Get(context.TODO(), namespacedName, currentPod); err != nil {
+						log.Error(err, "unable to get pod", pod.Name)
+					}
+
+					log.Error(err, "error waiting for PodRunning phase", currentPod.Name, currentPod.Status.Phase)
```

Comment:
```suggestion
					log.Error(err, "timed out waiting for pod to enter PodRunning phase", "podName", currentPod.GetName())
```

We don't need the extra call to `Get` here. If there's an error, just log it and exit the goroutine. The `Status.Phase` in the error message is helpful for devs debugging, but not necessary for the end user